File: java-webmvc-jpa/src/main/java/com/github/anddd7/entity/Product.java
package com.github.anddd7.entity;

import java.math.BigDecimal;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "products")
public class Product {

  @Id
  private Integer id;
  private String name;
  private BigDecimal price;

  public Integer getId() {
    return id;
  }

  public void setId(Integer id) {
    this.id = id;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public BigDecimal getPrice() {
    return price;
  }

  public void setPrice(BigDecimal price) {
    this.price = price;
  }
}

File: kotlin-webmvc-jpa/src/main/kotlin/com/github/anddd7/dto/Product.kt
package com.github.anddd7.dto

import java.math.BigDecimal
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Table

@Entity
@Table(name = "products")
data class Product(
    @javax.persistence.Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int = 0,
    val name: String,
    val price: BigDecimal
)

data class ProductStockDTO(
    val id: Int = 0,
    val name: String,
    val price: BigDecimal,
    val stock: BigDecimal
) {
    constructor(product: Product, stock: BigDecimal) : this(product.id, product.name, product.price, stock)
}
